### PR TITLE
Feature/9 option menu

### PR DESCRIPTION
**Pull request by aurelenc**
Depends on #104

### CODE REVIEW

File: src/game/components/Rectangle.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+** EPITECH PROJECT, 2022
+** Bomberman-Global-Offensive
+** File description:
+** Rectangle
+*/
+
+#include "ecs/Component.hpp"
+#include "raylib/shapes/Rectangle.hpp"
+
+#pragma once
```
Comment (AndreasLrx) [inline src/game/components/Rectangle.hpp:11]: We do not use pragma once in this project.

File: src/game/components/Size.hpp [hunk 2]
```diff
@@ -11,13 +11,14 @@
...
 
-namespace game
+namespace game::components
 {
     struct Size : public ecs::Component, public raylib::core::Vector3 {
-        Size(float px = 0, float py = 0, float pz = 0) : Vector3(px, py, pz) {}
-        Size(Vector3 vector) : Vector3(vector) {}
-        Size(Vector2 vector) : Vector3(vector) {}
+        Size(const raylib::core::Vector2 &psize) : raylib::core::Vector3(psize) {}
+        Size(const raylib::core::Vector3 &psize) : raylib::core::Vector3(psize) {}
```
Comment (AndreasLrx) [inline src/game/components/Size.hpp:18]: No need to use references for vectors.

**Review by AndreasLrx — CHANGES_REQUESTED** [review]
Create a Worlds class or something like it instead of using the Game class to manage worlds.
The Game class is intended to represent a game party only, and doesn't care about the settings and all other worlds. (That's why it contains a Map member)

Lot of small changes.

**Review by AndreasLrx — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Review by francoisbaudouin — APPROVED** [review]
ok

File: src/game/components/Image.hpp [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+** File description:
+** Image
+*/
+
 #ifndef GAME_COMPONENTS_IMAGE_HPP_
 #define GAME_COMPONENTS_IMAGE_HPP_
 
 #include <filesystem>
 #include "ecs/Component.hpp"
 #include "raylib/images/Image.hpp"
```
Comment (francoisbaudouin) [inline src/game/components/Image.hpp:13]: Does it exist ? it's not raylib/textures/Image.hpp ?

Comment (aurelenc) [reply src/game/components/Image.hpp:13]: raylib/images/Image.hpp for the raylib object
game/components/Image.hpp for the component

**Review by MisterPeModder — APPROVED** [review]
LGTM

**Pull request merged**